Mathlib/MeasureTheory/Measure/Haar/DistribChar.lean

```diff
@@ -0,0 +1,86 @@
+/-
+Copyright (c) 2024 Andrew Yang, Yaël Dillies, Javier López-Contreras. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Andrew Yang, Yaël Dillies, Javier López-Contreras
+-/
+import Mathlib.MeasureTheory.Measure.Haar.Unique
+
+/-!
+# The distributive character of Haar measures
+
+Given a group `G` acting on an measurable additive commutative group `A`, and an element `g : G`,
+one can pull back the Haar measure `μ` of `A` along the map `(g • ·) : A → A` to get another Haar
+measure `μ'` on `A`. By unicity of Haar measures, there exists some nonnegative real number `r` such
+that `μ' = r • μ`. We can thus define a map `distribHaarChar : G → ℝ≥0` sending `g` to its
+associated real number `r`. Furthermore, this number doesn't depend on the Haar measure `μ` we
+started with, and `distribHaarChar` is a group homomorphism.
+
+## See also
+
+[Zulip](https://leanprover.zulipchat.com/#narrow/channel/217875-Is-there-code-for-X.3F/topic/canonical.20norm.20coming.20from.20Haar.20measure/near/480050592)
+-/
+
+open MeasureTheory.Measure
+open scoped NNReal Pointwise ENNReal
+
+namespace MeasureTheory
+variable {G A : Type*} [Group G] [AddCommGroup A] [DistribMulAction G A] [MeasurableSpace A]
+  -- We only need `MeasurableConstSMul G A` but we don't have this class. So we erroneously must
+  -- assume `MeasurableSpace G` + `MeasurableSMul G A`
```

Comment:
This has already showed up in another PR. I don't understand this comment: G acts by continuous maps thanks to `[ContinuousConstSMul G A]`, so in particular it acts by measurable maps, so why do you need `[MeasurableSMul G A]`?

Reply:
Oh good point.

But still there is an issue: I can't use all the lemmas like `Measurable.const_smul` because they assume `MeasurableSMul G A`. By what you said, they are also true assuming `ContinuousConstSMul G A`, but no lemma applies to this situation.

I believe this is a good argument for introducing `MeasurableConstSMul`. What do you think?

Reply:
Yes, definitely!

Reply:
There you go! #24317

Reply:
Thanks, but doesn't build :-)